### PR TITLE
Add github action for uploading codecov results

### PR DESCRIPTION
**Pull request by rejas**
I added codecov testing coverage a while ago. This PR would now add a github action that uploads the results to codecov io

It is only a draft since:

a) I dont know if @MichMich wants it :-) PLus he would need to register the MM repo with codecov.io

b) I dont know on what occasion we should upload the results? Every push? Or only on merges?

c) Other comments?

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+fixes:
+  - "/home/runner/work/rejas/MagicMirror/::"
```
Comment (rejas) [inline codecov.yml:2]: This woul dhave to be adjusted to the original repo

**Comment by MichMich** [comment]
I like the idea. But I don't think this will help making the tests more complete. This will only show how awful the current test coverage is. 😂

Speaking of: the test suite could be so much improved. Currently for every test we start electron, while in most cases we could just "virtually" start the module and test what getDom returns.

File: codecov.yml (hunk 1 above)
Comment (fewieden) [reply codecov.yml:2]: @rejas I liked the idea of the automatic coverage check and added it to one of my modules. I first tried what you did and I could find online, but couldn't make it work properly.

Then I switched to using the action provided by codecov itself, where I didn't need to specify the path override.

https://github.com/fewieden/MMM-soccer/blob/72f1ef0ba84ece103e639faaadcdfaf1676c0ae6/.github/workflows/coverage.yml#L30-L48

Comment (rejas) [reply codecov.yml]: Thanks for the hint @fewieden I switched the PR to use the codecov action. Any more helpful hints?

**Comment by MichMich** [comment]
Good idea!

**Comment by MichMich** [comment]
Do you have some info on what I have to do to make it work on the mm repo?

**Comment by rejas** [comment]
> Do you have some info on what I have to do to make it work on the mm repo?

Only register the repo (and yourself via github) on https://codecov.io/

**Comment by MichMich** [comment]
Done. Hope I did it correctly. :)

**Comment by rejas** [comment]
Maybe we can take add the badge from codecov to the README. You should find under https://app.codecov.io/gh/MichMich/MagicMirror/settings/badge so we can see if it works

**Comment by fewieden** [comment]
Badge this is the badge for develop

Badge this is the badge for master

**Comment by MichMich** [comment]
Added it to the readme. Used the master because this will end up in the readme on the master branch. Or is there a markdown template thing to use the branch you are currently viewing?

**Pull request merged**